composer/loggers/mosaicml_logger.py

```diff
@@ -80,21 +83,69 @@ def __init__(
                 log.warning(f'Environment variable `{RUN_NAME_ENV_VAR}` not set, so MosaicMLLogger '
                             'is disabled as it is unable to identify which run to log to.')
                 self._enabled = False
-
+            
     def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
         self._log_metadata(hyperparameters)
 
     def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
         self._log_metadata(metrics)
 
     def after_load(self, state: State, logger: Logger) -> None:
+        # Log model data downloaded and initialized for run events
+        self._log_metadata({'model_initialized_time': time.time()})
         # Log WandB run URL if it exists. Must run on after_load as WandB is setup on event init
         for callback in state.callbacks:
             if isinstance(callback, WandBLogger):
                 run_url = callback.run_url
                 if run_url is not None:
                     self._log_metadata({'wandb/run_url': run_url})
 
+    def fit_start(self, state: State, logger: Logger) -> None:
+        # adds state to calculate training progress on log metadata
+        self._state = state
+
+    def _get_training_progress_metrics(self, state: State) -> Dict[str, Any]:
+        """Calculates training progress metrics.
+
+        If user submits max duration:
+        - in tokens -> format: [token=x/xx]
+        - in batches -> format: [batch=x/xx]
+        - in epoch -> format: [epoch=x/xx] [batch=x/xx] (where batch refers to batches completed in current epoch)
+        If batches per epoch cannot be calculated, return [epoch=x/xx]
+
+        If no training duration given -> format: ''
+        """
+        if state.max_duration is None:
```

Comment:
in what case will max_duration will be None?

Reply:
how come we return `{'training_progress': ''}` if there is no max_duration, but we return an empty dictionary (instantiated on L132) if the unit is not token, epoch, or batch? would we want these to be the same?

Reply:
I'm pretty confident max duration will default to 1 epoch for finetuning, but not sure for other runs so put a catch to make sure it doesn't error

Reply:
Note that max duration has to be set at this point -- it is type Union None as it does not need to be set on `Trainer.__init__` and can instead be set on `Trainer.fit()`. 

You could just assert it is not None here for type checker as this is a valid logical assumption (see L1777 Trainer for validation logic)